### PR TITLE
fix: deep clone initialValues passed to Form

### LINKED ISSUE #33981: Form.List modified the raw value while "preserve" false
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-form-list-bug-26qw4)

### Steps to reproduce
1. pass initialValue to Form and set preserve false
2. open modal -> close modal -> reopen

### What is expected?
field "users" was emptied, and the raw value was modified

### What is actually happening?
the raw value will never be modified

| Environment | Info |
|---|---|
| antd | 4.18.6 |
| React | 16.14.0 |
| System | Windows |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by nieyuyao**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

Fix #33981 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/form/__tests__/index.test.js [hunk 1]
```diff
@@ -134,6 +134,32 @@ describe('Form', () => {
 
       jest.useRealTimers();
     });
+    it('should not change `initialValues` passed to `Form`', async () => {
```
Comment (yoyo837) [inline components/form/__tests__/index.test.js:137]: 这个用例和你的修复似乎毫无关系呀。

**Comment by MadCcc** [comment]
It's more like a bug of rc-field-form. I don't think deep clone is a good solution.

**Comment by nieyuyao** [comment]
> It's more like a bug of rc-field-form.

🤔 Umm... When the component is unmounted, the fields of `initialValues ` are deleted.

**Comment by MadCcc** [comment]
Fixed in https://github.com/react-component/field-form/pull/370

**Pull request closed without merging**